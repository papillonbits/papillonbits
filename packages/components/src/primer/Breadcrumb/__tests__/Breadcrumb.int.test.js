import { items, regular, inactive } from './Breadcrumb.int.story'

jest.mock('@papillonbits/library/array', () => ({
  getIndexItems: () => {},
  getIndexItemsWithSelected: () => {},
}))

const libraryArrayMockObject = require('@papillonbits/library/array')

describe('<Breadcrumb />', () => {
  const indexItemsDataObject = [
    { ...items[0], index: 0 },
    { ...items[1], index: 1 },
    { ...items[2], index: 2 },
  ]

  const newIndexItemsWithSelectedDataObject = [
    { ...items[0], index: 0, isSelected: true },
    { ...items[1], index: 1, isSelected: false },
    { ...items[2], index: 2, isSelected: false },
  ]

  beforeEach(() => {
    jest.spyOn(libraryArrayMockObject, 'getIndexItems').mockReturnValue(indexItemsDataObject)

    jest.spyOn(libraryArrayMockObject, 'getIndexItemsWithSelected').mockReturnValue(newIndexItemsWithSelectedDataObject)
  })

  afterEach(() => jest.clearAllMocks())

  describe('Render', () => {
    test('must match regular()', () => {
      expect(global.renderToJSON(regular())).toMatchSnapshot()
    })

    test('must match inactive()', () => {
      expect(global.renderToJSON(inactive())).toMatchSnapshot()
    })
  })
})
