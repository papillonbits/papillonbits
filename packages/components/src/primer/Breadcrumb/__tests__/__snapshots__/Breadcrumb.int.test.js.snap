// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Breadcrumb /> Render must match inactive() 1`] = `
<div>
  <nav
    aria-label="Breadcrumb"
    class="breadcrumb-inactive"
  >
    <ol>
      <li
        class="breadcrumb-item"
        href="/fantastic-folder"
      >
        <a
          href="/fantastic-folder"
        >
          Fantastic Folder
        </a>
      </li>
      <li
        class="breadcrumb-item"
        href="/fantastic-folder/another-fantastic-folder"
      >
        <a
          href="/fantastic-folder/another-fantastic-folder"
        >
          Another Fantastic folder
        </a>
      </li>
      <li
        aria-current="page"
        class="breadcrumb-item breadcrumb-item-selected text-gray"
        href=""
      >
        One More Fantastic Folder
      </li>
    </ol>
  </nav>
</div>
`;

exports[`<Breadcrumb /> Render must match regular() 1`] = `
<div>
  <nav
    aria-label="Breadcrumb"
    class=""
  >
    <ol>
      <li
        class="breadcrumb-item"
        href="/fantastic-folder"
      >
        <a
          href="/fantastic-folder"
        >
          Fantastic Folder
        </a>
      </li>
      <li
        class="breadcrumb-item"
        href="/fantastic-folder/another-fantastic-folder"
      >
        <a
          href="/fantastic-folder/another-fantastic-folder"
        >
          Another Fantastic folder
        </a>
      </li>
      <li
        aria-current="page"
        class="breadcrumb-item breadcrumb-item-selected text-gray"
        href=""
      >
        One More Fantastic Folder
      </li>
    </ol>
  </nav>
</div>
`;
